source/simulator/core.cc

```diff
@@ -999,6 +1010,18 @@ namespace aspect
       // needed.  All other matrix blocks are left empty here.
       if (have_fem_compositional_field)
         coupling[x.compositional_fields[0]][x.compositional_fields[0]] = DoFTools::always;
+
+      // If we are using VolumeOfFluid interface tracking, create a matrix block in the
+      // field corresponding to the volume fraction.
```

Comment:
This comment does not fully explain why you need this. Why can you not use the same matrix block that is usually used for compositional fields? Is that one already used for something else (some of the other parameters you use)? If not, we might not need this matrix block and could instead use the existing one. We implemented this a while ago, because having many compositional fields can become the main memory limitation for a model. Please update the comment.

Reply:
I have now implemented an approach similar to the current one for the compositional fields, but using the VoF 0 matrix.

Reply:
Very nice! Thanks!